Automattic/jetpack
# Remove convert products data out of redux into tanstack

## Proposed changes:

* Move over product data, product activation, and standalone product installation to tanstack query
* Remove redux code pertaining to Products
* Remove deactivate/uninstall functionality from My Jetpack since it is no longer possible to do that from the UI
* Overall goal is to make the behavior exactly as it was in redux
* Split up `useProduct` into product data, activation, and installation due to useMutation loading state issues

Note: useMutation is a hook, and therefore cannot be called conditionally. Putting this into `useProduct` was creating an instance of the mutation every time whether it was used or not. It seemed more efficient to set it up in a separate hook so that it is only created/used when it's needed

### Other information:


## Jetpack product discussion

P2: p1HpG7-rcF-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin (hard, but not impossible, to test plugin installation on local dev environment, easiest to just do it on the beta plugin)
2. Install but DO NOT ACTIVATE the Protect and VideoPress standalone plugins
3. Purchase a Boost subscription for the site, but do not install the plugin
4. Go to `/wp-admin/admin.php?page=my-jetpack`
5. Make sure the status's look correct
![image](https://github.com/Automattic/jetpack/assets/65001528/a915d7aa-112e-426f-9937-06b6cbc5de24)
6. Click the "Install Plugin" CTA on the Boost card. Make sure the loading state looks good/correct, and that the boost plugin is installed
![image](https://github.com/Automattic/jetpack/assets/65001528/4beb1717-39d4-48f3-8b06-8aff03928760)
![image](https://github.com/Automattic/jetpack/assets/65001528/43229d1e-fa64-4c6c-958a-7cb10abff63b)
7. Deactivate and uninstall the Boost plugin, then go to `https://totally-secret-wolf.jurassic.ninja/wp-admin/admin.php?page=my-jetpack#/add-boost`
![image](https://github.com/Automattic/jetpack/assets/65001528/3cd282e5-2ace-4274-ae3a-b6cf04e48c7a)
NOTE: You'll see that the CTA says "Get Jetpack Boost" not "Install Jetpack Boost". This is a bug currently on trunk, and due to the size of this PR already, I chose not to fix it here. I have it logged as a task to fix on the My Jetpack board
8. Click the `Get Jetpack Boost` CTA, you will be directed to checkout for a renewal (part of the bug mentioned above), go back to your site and make sure the Boost plugin was installed
![image](https://github.com/Automattic/jetpack/assets/65001528/385aaece-8d27-47a9-9514-6f366396f9f1)
9. On the Protect card, click Activate. Make sure the loading state looks good/correct and that the plugin gets activated
![image](https://github.com/Automattic/jetpack/assets/65001528/20ab2d2e-ca9d-4219-9875-cfc8dd349f57)
10. On the VideoPress card, click "Learn More". Make sure the interstitial says "Get Jetpack VideoPress"
![image](https://github.com/Automattic/jetpack/assets/65001528/20a7b862-9e17-4fa1-a9d2-e2acc009ec6d)
11. Click the CTA, you don't need to purchase the plan. When you get back to your site, make sure the standalone plugin was activated
![image](https://github.com/Automattic/jetpack/assets/65001528/a936cccc-2cd5-44c1-9a98-3ddb38de795e)
12. The functionality that was migrated in this PR had to do with Activating/Installing standalone products, as well as getting all product data. If there are any places that product data is displayed or needed, or ways to install/activate plugins that I am not aware of, check those as well to make sure they work correctly. I did search the code extensively, so there should not be any places unaccounted for but you never know.
